killbill-integration-tests/core/test_mixed_catalog_with_usage.rb

```diff
@@ -0,0 +1,311 @@
+      all_invoices = @account.invoices(true, @options)
+      assert_equal(2, all_invoices.size)
+      sort_invoices!(all_invoices)
+      second_invoice = all_invoices[1]
+      check_invoice_no_balance(second_invoice, 0, 'USD', '2013-10-01')
+      check_invoice_item(second_invoice.items[0], second_invoice.invoice_id, 0, 'USD', 'USAGE', 'voip-monthly-by-usage', 'voip-monthly-by-usage-evergreen', '2013-09-01', '2013-10-01')
+      # AGGREGATE mode by default
+      check_invoice_consumable_item_detail(second_invoice.items[0],
+                                           [{:tier => 1, :unit_type => 'minutes', :unit_qty => 0, :tier_price => 0.99 }], 0)
+
+      # 2013-10-15
+      kb_clock_add_days(14, nil, @options)
+
+      # Add usage for the month
+      usage_input = [{:unit_type => 'minutes',
+                      :usage_records => [{:record_date => '2013-10-01', :amount => 1},
+                                         {:record_date => '2013-10-02', :amount => 1},
+                                         {:record_date => '2013-10-03', :amount => 1},
+                                         {:record_date => '2013-10-04', :amount => 1},
+                                         {:record_date => '2013-10-05', :amount => 1},
+                                         {:record_date => '2013-10-07', :amount => 1}]
+                     }]
+      record_usage(bp.subscription_id, usage_input, @user, @options)
+
+      # Verify IMMEDIATE cancellation
+      requested_date = nil
+      entitlement_policy = nil
+      billing_policy = nil
+      use_requested_date_for_billing = nil
+      bp.cancel(@user, nil, nil, requested_date, entitlement_policy, billing_policy, use_requested_date_for_billing, @options)
+
+      bp = get_subscription(bp.subscription_id, @options)
+      check_subscription(bp, 'Voip', 'BASE', 'NO_BILLING_PERIOD', 'DEFAULT', DEFAULT_KB_INIT_DATE, '2013-10-15', DEFAULT_KB_INIT_DATE, '2013-10-15')
+      assert_equal('CANCELLED', bp.state)
+
+      wait_for_expected_clause(3, @account, @options, &@proc_account_invoices_nb)
+
+      # Third invoice
+      all_invoices = @account.invoices(true, @options)
+      assert_equal(3, all_invoices.size)
+      sort_invoices!(all_invoices)
+      third_invoice = all_invoices[2]
+      check_invoice_no_balance(third_invoice, 5.94, 'USD', '2013-10-15')
+      check_invoice_item(third_invoice.items[0], third_invoice.invoice_id, 5.94, 'USD', 'USAGE', 'voip-monthly-by-usage', 'voip-monthly-by-usage-evergreen', '2013-10-01', '2013-10-15')
+      # AGGREGATE mode by default
+      check_invoice_consumable_item_detail(third_invoice.items[0],
+                                           [{:tier => 1, :unit_type => 'minutes', :unit_qty => 6, :tier_price => 0.99 }], 5.94)
+
+      # 2013-12-15
+      kb_clock_add_months(2, nil, @options)
+
+      # No new invoice is generated
+      assert_equal(3, @account.invoices(true, @options).size)
+    end
+
+    # Basic test to verify/understand the catalog
```

Comment:
Not sure exactly what this comment on each test mean ❓

Reply:
The main goal of this test file is to verify the behavior when switching between these two plans (tests that I'm currently working on). These ones just verify the behavior of the individual plans in isolation.